pkg/scheduler/testing/fake_lister.go

```diff
@@ -53,11 +55,16 @@ func (f FakePodLister) FilteredList(podFilter algorithm.PodFilter, s labels.Sele
 	return selected, nil
 }
 
-var _ algorithm.ServiceLister = &FakeServiceLister{}
+var _ corelisters.ServiceLister = &FakeServiceLister{}
```

Comment:
we should be able to remove the fakes as well, tests can use fake clientset to create a sharedinformerfactory.

Reply:
SG

Reply:
After some digging, I think we can't remove the `Fake.*Lister` since some of the test depends on them, ex:

```go
// FakePodLister implements PodLister on an []v1.Pods for test purposes.
type FakePodLister []*v1.Pod

// List returns []*v1.Pod matching a query.
func (f FakePodLister) List(s labels.Selector) (selected []*v1.Pod, err error) {
	for _, pod := range f {
		if s.Matches(labels.Set(pod.Labels)) {
			selected = append(selected, pod)
		}
	}
	return selected, nil
}

// FilteredList returns pods matching a pod filter and a label selector.
func (f FakePodLister) FilteredList(podFilter algorithm.PodFilter, s labels.Selector) (selected []*v1.Pod, err error) {
	for _, pod := range f {
		if podFilter(pod) && s.Matches(labels.Set(pod.Labels)) {
			selected = append(selected, pod)
		}
	}
	return selected, nil
}
```

Reply:
However, I try to remove every `Fake.*Lister` which created with an empty array of objects.

Reply:
I didn't all of them, just the ones related to this PR, is there a problem with removing the fake  ServiceLister, ControllerLister, ReplicaSetLister and PDBLister?

Reply:
Yes, we can't replace them with fake informer factory, because the tests build fake listers with an array of objects like this:

https://github.com/kubernetes/kubernetes/blob/ff7a09928b2c0071c93936ca7157ddc7b004d9fe/pkg/scheduler/algorithm/priorities/selector_spreading_test.go#L341-L345

The only fake lister we could remove is the FakePDBLister, please take a look at the lastest commit.

Reply:
Ok, sg